eip-0018.md

```diff
@@ -0,0 +1,161 @@
+ 
+ val campaignOut = OUTPUTS(2)
+ val validCampaign = campaignOut.tokens(0)._1 == tokenId && campaignOut.R4[Int].get == firstUnusedCampaignId
+ 
+
+ properControlBox && validTokens && validScript && validPayment && validCampaignIdUpdate && validCampaign
+}
+```
+
+So to register campaign, one need to pass control box as a data input, sell contract among inputs, and create a box with 
+campaign box data specified below in outputs, as well as an updated campaign token sale box.
+
+
+Control box NFT id: 72c3fbce3243d491d81eb564cdab1662b1f8d4c7e312b88870cec79b7cfd4321
+Tokensale box NFT id: 15b0ae41c24230069ff96dacbac0932850ac0c2a0924daf72a39e88cbcf3acd5
+Campaign identification token: 05b66b97e5802f6447b67fe30cb4055e14d6b17bb14f5f563d65c9622c43a659
+
+Tokensale box P2S address: yvNXjWe8vBvZTXwyUHemPU59CRfm8AnvnzXowRQkQ9hoiGXsS7oEUGLPof6RoYAdKXEgTWR4qTK44w3GBuNdBjXeWcHFeBvyFxHWdEBsnqnYhtaFC2S71eHRPpLEndDghZebLdx25nufFh8YFJ9D8gTxTvxqahBgzpJT7pbUAUEkE2iRSZwpiXvj5SUmA6vmTrQzMTxicBG2mGV1NLq5rjBCAxjM5FpNSJ1KgqrxeTAfBvs1QnMUZ6CTBtGPLyNUpFoPTaYUPp9PfBni7FAQsumx36tukUHA2p3NdgfhTbYYSJTDQi7eWW2YikFvczXpTCfecbCL7HA1o6Cg6DvntmpJHWfjopWnsbpDdLg6SbP4Fup7wCnQaeL6NcSTVP1K5btmK5JryZc36V8KdHkheyaVzYTMA1Gufp6bAGtZP
+
+## Campaign Box Data
+
+For registering campaign, it is enough just to create box with ErgoFund campaign token, proper campaign ID, campaign desc, script, recommended deadline, min to raise. Then offchain scanner will find it (and so UI will be able to display campaign data).
+
+For the scanner, there are following requirements campaign box should satisfy:
+
+value >= 1 ERG (1000000000 nanoERG)
+script = "4MQyMKvMbnCJG3aJ" (false proposition, so no one can spend)
+
+Registers: 
+*R4* - campaign ID (Int)
+*R5* - campaign desc (byte array)
+*R6* - campaign script (funds collected will go to this)
+*R7* - fundraising deadline (Int, # of block, exclusive) 
+*R8* - min value for successful fundraising (Long)  
+
+https://explorer.ergoplatform.com/en/transactions/2e25bc0ea4d01108ab1cd76969f49022228b533a2ea50540f6cde6258029a510
+
+
+example:
+
+```json
+[
+  {
+    "address": "4MQyMKvMbnCJG3aJ",
+    "value": 100000000,
+    "assets": [
+      {
+        "tokenId": "05b66b97e5802f6447b67fe30cb4055e14d6b17bb14f5f563d65c9622c43a659",
+        "amount": 1
+      }
+    ],
+    "registers": {
+      "R4": "0400",
+      "R5": "0e00",
+      "R6": "08cd0327e65711a59378c59359c3e1d0f7abe906479eccb76094e50fe79d743ccc15e6",
+      "R7": "04a0be49",
+      "R8": "0580d0acf30e"
+    }
+  }
+]
+```
+
+## Pledge Contract
+
+Pledge contract is an example of self-sovereign DeFi principle. The contract is allowing box being protected by it to be spent if a spending transaction can collect at least *minToRaise* ERGs to *projectPubKey*, and inclusion height of the transaction is less than *deadline*. Otherwise, after *deadline* height is met, funds can be claimed by *projectPubKey*. Please note that both *backerPubKey* and
```

Comment:
I think funds can be claimed by *projectPubKey* Should be, funds can be claimed by *backerPubKey*